editor/components/block-settings-menu/html-converter.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { IconButton, withAPIData } from '@wordpress/components';
+import { rawHandler, getBlockContent } from '@wordpress/blocks';
+import { compose } from '@wordpress/element';
+import { withSelect, withDispatch } from '@wordpress/data';
+
+export function HTMLConverter( { block, onReplace, small, user, role } ) {
+	if ( ! block || block.name !== 'core/html' ) {
+		return null;
+	}
+
+	const label = __( 'Convert to blocks' );
+
+	const convertToBlocks = () => {
+		onReplace( block.uid, rawHandler( {
+			HTML: getBlockContent( block ),
+			mode: 'BLOCKS',
+			canUserUseUnfilteredHTML: get( user, [ 'data', 'capabilities', 'unfiltered_html' ], false ),
+		} ) );
+	};
+
+	return (
+		<IconButton
+			className="editor-block-settings-menu__control"
+			onClick={ convertToBlocks }
+			icon="screenoptions"
+			label={ small ? label : undefined }
+			role={ role }
+		>
+			{ ! small && label }
+		</IconButton>
+	);
+}
+
+export default compose(
+	withSelect( ( select, { uid } ) => {
+		const { getBlock, getCurrentPostType } = select( 'core/editor' );
+		return {
+			block: getBlock( uid ),
+			postType: getCurrentPostType(),
+		};
+	} ),
+	withDispatch( ( dispatch ) => ( {
+		onReplace: dispatch( 'core/editor' ).replaceBlocks,
+	} ) ),
+	withAPIData( ( { postType } ) => ( {
```

Comment:
We are trying to get rid of `withAPIData` usage and deprecate it as explained in #7390 and #7397. An alternative approach would be to use `_links` object returned from the REST API as implemented in other places by @danielbachhuber, examples: #7495 or even better #6724.

Reply:
It can be done in a separate PR given that there are a few occurrences of `get( user, [ 'data', 'capabilities', 'unfiltered_html' ], false )`, but this should be started in parallel.

Reply:
Also related #6361

Reply:
Fixed in 138c7f1.

Also added the `canUserUseUnfilteredHTML` selector in d0e747e to substitute the rest of occurrences in a separate PR once this lands.

Reply:
https://github.com/WordPress/gutenberg/pull/7729 substitutes the remaining cases of `unfiltered_html` and `withAPIData` and depends on this being merged.